### PR TITLE
dev-yoshiki to master

### PR DESCRIPTION
**Pull request by YoshikiOhtani**
I created a custom trigger container so that the `mjd`, `millisec`, `nanosec` parameters are obtained to keep the precision of the timestamps for the event coincidence algorithm. Then I updated some functions to store the parameters to the container. The uproot module dependency (requiring v0.4.x) is also updated.

### CODE REVIEW

**Comment by YoshikiOhtani** [comment]
Sorry now I changed the destination of merge branch to master since Alessio suggested me to do so. It seems all the commits to update-ctapipe-v0.12 are merged to this branch, and so this pull request means that the modules in master branch can be only used in ctapipe v0.12.

File: ctapipe_io_magic/__init__.py [hunk 1]
```diff
@@ -32,7 +27,7 @@
     SimulationConfigContainer,
     PointingContainer,
     TelescopePointingContainer,
-    TelescopeTriggerContainer,
+    # TelescopeTriggerContainer,
```
Comment (aleberti) [inline ctapipe_io_magic/__init__.py:30]: Simply remove this if not needed, no need to keep it as a comment.

File: ctapipe_io_magic/__init__.py [hunk 2]
```diff
@@ -1012,10 +1018,14 @@ def _mono_event_generator(self, telescope):
...
-                data.trigger.tel[tel_i + 1] = TelescopeTriggerContainer(time=Time(
-                    time_tmp, format='unix', scale='utc', precision=9))
+                time_tmp = Time(event_data['MJD'], scale='utc', format='mjd')
+
+                data.trigger.tel[tel_i + 1] = TelescopeTriggerContainer(
+                    time=Time(time_tmp, format='unix', scale='utc', precision=9),
+                    mjd=event_data['mjd'],
+                    millisec=event_data['millisec'],
+                    nanosec=event_data['nanosec']
+                )
```
Comment (aleberti) [inline ctapipe_io_magic/__init__.py:1028]: Please do the same modifications also in _pedestal_event_generator.

**Review by aleberti — CHANGES_REQUESTED** [review]

**Comment by YoshikiOhtani** [comment]
Hi @aleberti, thanks for the review, and I have just pushed a commit. Not only the pedestal event generator but I also modify the stereo event generator just in case, though I have not yet tested them. I will inform you if I find some problems.

By the way I think in general we can write the code more simply, but since we don't have time until the software school, I will try to simply them after the school.

**Comment by aleberti** [comment]
> Hi @aleberti, thanks for the review, and I have just pushed a commit. Not only the pedestal event generator but I also modify the stereo event generator just in case, though I have not yet tested them. I will inform you if I find some problems.

Good, thank you 

> By the way I think in general we can write the code more simply, but since we don't have time until the software school, I will try to simply them after the school. 

Indeed yesterday I opened an issue on this #11. I will work on it in parallel to the preparation for the LST analysis school, I already have some code snippets I tested in the last days.

**Review by aleberti — APPROVED** [review]

**Comment by YoshikiOhtani** [comment]
Hi @aleberti, thanks for the approval but can you please wait a bit to merge? Because I'm now implementing an idea to keep a precision of timestamps without creating the custom trigger container. I will push a new commit soon, so it would be appreciated if you could review it!

**Comment by aleberti** [comment]
Tests are passing. I will merge the PR.

**Comment by aleberti** [comment]
> Hi @aleberti, thanks for the approval but can you please wait a bit to merge? Because I'm now implementing an idea to keep a precision of timestamps without creating the custom trigger container. I will push a new commit soon, so it would be appreciated if you could review it!

sorry, I was too fast. but don't worry, simply open a new PR, I think it's also better to keep those separated since you are introducing a new implementation.

**Comment by YoshikiOhtani** [comment]
You are faster :), but ok, I will open a new pull request.

**Pull request merged**